src/ui/mainwindow.cpp

```diff
@@ -468,6 +514,22 @@ void MainWindow::createTab(int index)
     m_tabBar->setCurrentIndex(index);
 }
 
+void MainWindow::duplicateTab(int index)
```

Comment:
Let's make `index` a required parameter here, and actually duplicate the specific tab. That will allow duplicating tabs from a context menu, once we have it.